src/components/Vote/index.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import connect from 'utils/connect';
+import { voteOnArticle, initArticleVoteCount } from 'modules/vote';
+import Icon from 'components/Icon';
+
+import './Vote.scss';
+
+/**
+ * Component for article vote
+ * @summary composes the upvote and the downvote icons along with the vote count
+ */
+class Vote extends React.Component {
+  /**
+    * @name Vote propTypes
+    * @type {propTypes}
+    *
+    * @param {Object} props - React PropTypes
+    *
+    * @property {string} articleId -id of the article to vote on
+    * @property {string} activeVoteType - article voteType by the current user
+    * @property {Function} voteOnArticle -action creator to vote on an article
+    * @property {Function} articleVote -redux state map to props
+    *
+    * @return {Array} React propTypes
+    */
+  static propTypes = {
+    articleId: PropTypes.string.isRequired,
+    activeVoteType: PropTypes.string,
+    voteOnArticle: PropTypes.func.isRequired,
+    initArticleVoteCount: PropTypes.func.isRequired,
+    articleVote: PropTypes.isRequired,
+    upVoteCount: PropTypes.number,
+    downVoteCount: PropTypes.number
+  }
+
+  static defaultProps = {
+    activeVoteType: null,
+    upVoteCount: 0,
+    downVoteCount: 0
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = { activeVoteType: props.activeVoteType };
+  }
+
+  componentDidMount() {
+    const { initArticleVoteCount: articleVoteCount, upVoteCount, downVoteCount } = this.props;
+
+    articleVoteCount({ upVoteCount, downVoteCount });
+  }
+
+  /**
+    * handles vote action
+    * @method
+    *
+    * @param {Object} payload
+    * @param {string} payload.voteType - type of vote
+    *
+    * @return {void}
+    */
+  onVoteHandler = ({ voteType }) => {
+    const { voteOnArticle: voteOnArticleHandler, articleId } = this.props;
+    const { activeVoteType } = this.state;
+
+    const vote = voteType === activeVoteType ? 'nullVote' : voteType;
+
+    voteOnArticleHandler({ articleId, voteType: vote, prevVote: activeVoteType });
+
+    this.setState((prevState) => ({ ...prevState, activeVoteType: vote }));
```

Comment:
We don't need `...prevState`, React does this by default. Or is this a new API?
```suggestion
    this.setState({ activeVoteType: vote });
```